src/tree/updater_gpu_hist.cu

```diff
@@ -43,14 +43,15 @@ using GradientPairSumT = GradientPairPrecise;
  * \param temp_storage Shared memory for intermediate result.
  */
 template <int BLOCK_THREADS, typename ReduceT, typename TempStorageT>
-__device__ GradientPairSumT ReduceFeature(const GradientPairSumT* begin,
-                                          const GradientPairSumT* end,
+__device__ GradientPairSumT ReduceFeature(common::Span<const GradientPairSumT> feature_histogram,
                                           TempStorageT* temp_storage) {
   __shared__ cub::Uninitialized<GradientPairSumT> uninitialized_sum;
   GradientPairSumT& shared_sum = uninitialized_sum.Alias();
 
   GradientPairSumT local_sum = GradientPairSumT();
   // For loop sums features into one block size
+  auto begin = feature_histogram.data();
```

Comment:
Actually we might be able to use non-pointer iteration with this line:
https://github.com/dmlc/xgboost/blob/be0bb7dd908ba910f73ee33095df915df6a0c138/src/common/span.h#L144
change to use `<`,
and remove the check in this function:
https://github.com/dmlc/xgboost/blob/be0bb7dd908ba910f73ee33095df915df6a0c138/src/common/span.h#L177

The error report will be delayed to applying `* operator` and `-> operator`, but it will provide us some flexibility.  WDYT?